src/rai_finetune/rai_finetune/data/langfuse_obs_extractor.py

```diff
@@ -0,0 +1,489 @@
+# Copyright (C) 2025 RAI Development Team
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Author: Julia Jia
+
+
+import argparse
+import json
+import logging
+import os
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional
+
+try:
+    from langfuse import Langfuse
+except ImportError:
+    print("Error: langfuse package not found. Install with: pip install langfuse")
+    exit(1)
+
+# Configure logging
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+)
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ObservationExtractionConfig:
+    """Configuration for observation extraction pipeline from Langfuse"""
+
+    langfuse_host: str = "http://localhost:3000"
+    langfuse_public_key: str = ""
+    langfuse_secret_key: str = ""
+
+    # Observation related
+    output_file: str = "observations.jsonl"
+    max_observations: int = 1000
+    type_filter: str = "GENERATION"  # e.g., GENERATION, EVENT, SPAN
```

Comment:
```suggestion
    type_filter: Literal["GENERATION", "EVENT", "SPAN" ] = "GENERATION"  # e.g., GENERATION, EVENT, SPAN
```

If I understand correctly there is a set list of valid options.

Reply:
Updated per your suggestion.